### PR TITLE
Design for shared bandwidth group solution

### PR DESCRIPTION
**Pull request by sergii-iakovenko**

### CODE REVIEW

File: docs/design/solutions/shared-bandwidth/shared-bandwidth.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+![Data Model changes diagram](domain-model-changes.png "Data Model changes diagram")
+
+From persistence perspective, "shared_bandwidth_group" is just an additional property on `flow_path` and `path_segment` graph nodes with corresponding indexes.
+
+### API
+No changes required in public API or internal messaging.
+
+### PCE changes
+Normally, PCE relies on ISL available bandwidth which is already precalculated. But in the case of flow reroute/update operations, PCE accepts a list of paths to reuse their bandwidth.
+The path grouping must be taken into account - bring back the allocated bandwidth only once for each group.
```
Comment (surabujin) [inline docs/design/solutions/shared-bandwidth/shared-bandwidth.md:49]: Sounds like existing logic "PCE accepts a list of paths to reuse their bandwidth" can be fully replaced with new logic based on "shared bandwidth groups".

File: docs/design/solutions/shared-bandwidth/shared-bandwidth.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+- A y-flow by design have 2 or more paths sharing ISLs. The common (shared) part of all sub-flows must occupy only 
+bandwidth requested by the y-flow. Without the shared bandwidth solution 2 sub-flows will allocate double bandwidth.
+
+## Solution overview
+The goal is to avoid repetitive bandwidth allocation on an ISL if 2 or more bound flow paths lay on it. 
+So, Kilda must group such bound paths. For this purpose, a new property ("shared_bandwidth_group") is to be introduced on the path and path segment levels. 
+
+The available bandwidth for an ISL is calculated by deducing from ISL capacity the allocated bandwidth of all related path segments:
+- With no grouping - ```available_bandwidth(ISL) = bandwidth(ISL) - ∑ bandwidth(PSₙ)```, where PS is a set of path segments that lay on the ISL.
+- With grouping - ```available_bandwidth(ISL) = bandwidth(ISL) - max(PS₁) - max(PS₂) - ... - max(PSₖ)```, where PS₁, PS₂ ... PSₖ are sets of path segments grouped by 1, 2, ... k correspondingly.
```
Comment (surabujin) [inline docs/design/solutions/shared-bandwidth/shared-bandwidth.md:28]: Is the field "shared_bandwidth_group_id" a mandatory field for flow path? I.e it must be filled for all paths? If so, the math should stay simple.

**Review by surabujin — APPROVED** [review]

File: docs/design/solutions/shared-bandwidth/shared-bandwidth.md (hunk 2 above)
Comment (sergii-iakovenko) [reply docs/design/solutions/shared-bandwidth/shared-bandwidth.md:28]: Right, the field will be filled for all paths. 
As for me, the math is pretty simple: group all, take the max value for each group, subtract from ISL bandwidth.

File: docs/design/solutions/shared-bandwidth/shared-bandwidth.md (hunk 1 above)
Comment (sergii-iakovenko) [reply docs/design/solutions/shared-bandwidth/shared-bandwidth.md:49]: PoC implementation showed that reusing paths is still required in PCE - swap endpoints operation needs it.

**Review by nikitamarchenko — APPROVED** [review]
LGTM

**Review by dpoltavets — APPROVED** [review]

**Pull request merged**